presbuzz-frontend/src/components/Analytics.js

```diff
@@ -0,0 +1,214 @@
+import Calendar from "./Calendar";
+import Mentions from "./Mentions";
+import Compare from "./Compare";
+import constants from "./constants";
+import "../stylesheets/Analytics.css";
+import CircularProgress from "@material-ui/core/CircularProgress";
+import ButtonBases from "./ButtonBases";
+import "../index.css";
+import Switch from "@material-ui/core/Switch";
+import FormGroup from "@material-ui/core/FormGroup";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import { Redirect, withRouter, Route } from "react-router-dom";
+import Articles from "./Articles";
+
+class Analytics extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      result: [],
+      showComponent: false,
+      checked: false,
+      redirect: false
+    };
+
+    this.handleClick = this.handleClick.bind(this);
+  }
+  componentWillMount() {
+    let term = this.props.term;
+    fetch(constants.BASE_URL + constants.TERM_PATH + term)
+      .then(response => response.json())
+      .then(data => this.setState({ result: data.articles }))
+      .then(data => this.setState({ showComponent: true }));
+  }
+  handleChange = name => event => {
+    this.setState({ checked: event.target.checked });
+  };
+
+  handleClick(event) {
+    this.setState({ redirect: true });
+  }
+  render() {
+    let style = {
+      height: "500px"
+    };
+    let style1 = {
+      height: "420vh"
+    };
+
+    let style2 = {
+      height: "320vh"
+    };
+    if (this.state.redirect === true) {
+      return (
+        <div>
+          <Route path="/articles" render={() => <Articles ids="1,2,3" />} />
+          <Redirect to={"/articles"} />
+        </div>
+      );
+    }
+
+    if (this.state.checked === true) {
+      if (this.state.showComponent === true) {
+        return (
+          <div className="bg" style={style2}>
+            <div className="row">
+              <div className="col-md-12" onClick={this.handleClick}>
+                <ButtonBases className="button" />
+              </div>
+            </div>
+            <div className="container">
+              <div className="row">
+                <div className="col-md-4 box">
+                  <FormGroup row>
+                    <FormControlLabel
+                      control={
+                        <Switch
+                          checked={this.state.checked}
+                          onChange={this.handleChange("checked")}
+                          value="checked"
+                        />
+                      }
+                      label="Compress"
+                    />
+                  </FormGroup>
+                  <Mentions
+                    mentions={Object.keys(this.state.result).length}
+                    size="50px"
+                  />
+                </div>
+                <div className="col-md-4 box">
+                  <h1 className="heading">Year Wise Mentions</h1>
+                  <Barchart data={this.state.result} />
```

Comment:
We need to parse `this.state.result` and send smaller objects to each component. Just the data that it needs.
Reason: this.state.result can be a huge object on production with thousands of articles in the result, Hence crashing our website.
**You can add this feature in next PR**